pkg/operator/configobserver/cloudprovider/observe_cloud_volume_plugin.go

```diff
@@ -0,0 +1,96 @@
+package cloudprovider
+
+import (
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/cloudprovider"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+// NewCloudVolumePluginObserver returns a new cloudprovider observer for syncing cloud provider specific
+// information to controller-manager, leaving only storage controller loops enabled.
+func NewCloudVolumePluginObserver(targetNamespaceName string, cloudVolumePluginPath, cloudProviderConfigPath []string) configobserver.ObserveConfigFunc {
+	cloudObserver := &cloudProviderObserver{
```

Comment:
Why are we reusing the cloudProviderObserver, would it not be better to have a dedicated observer for this CloudVolumePluginObserver?

Reply:
I'm reusing the tested part of that code. The behavior and argument types are very similar in both `NewCloudVolumePluginObserver` and `NewCloudProviderObserver`, and which method will be used is decided on this line https://github.com/openshift/library-go/pull/994/files#diff-e571c7ed0d7124cda88d781ed909db675ef38ce8c54b4cab07b1f3e0283a519bR22 This is more error-prone for something which will stay in a cluster for a couple of releases.

Reply:
How many repos will use this function?  Also, a separate package seems proper here.

Reply:
so, i'm a little confused. is it better to make a new copy here or am i misinterpreting  the code?